components/brave_ads/browser/ads_notification_handler.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/ads_notification_handler.h"
+
+#include <utility>
+
+#include "brave/components/brave_ads/browser/ads_service_impl.h"
+#include "chrome/browser/profiles/profile.h"
```

Comment:
use `BrowserContext` here - `Profile` is not allowed in `components`

Reply:
Would it not still be a layer violation given that all of the methods of `NotificationHandler` all take `Profile*` as well? Seems like I should move this file to ` browser/notifications` although that may present its own issues given the use of `AdsServiceImpl`.